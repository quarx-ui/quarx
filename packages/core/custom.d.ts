declare module '*.png';
declare module '*.md';
declare module '*.svg';
declare module '*.svg' {
    // eslint-disable-next-line import/newline-after-import
    import { SFC, SVGProps } from 'react';
    export const ReactComponent: SFC<SVGProps<SVGSVGElement>>;
    const src: string;
    export default src;
}

interface ObjectConstructor {
    keys<T>(obj: T): Array<Exclude<keyof T, symbol | number>>;
    entries<T>(obj: T): Array<[Exclude<keyof T, symbol | number>, T[keyof T]]>;
}
