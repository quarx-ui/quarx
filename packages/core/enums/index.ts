export * from './QxDevice';
export * from './QxSize';
export * from './QxBorderSize';
