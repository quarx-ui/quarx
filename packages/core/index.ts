import './polyfills';

// WARNING: Изменение порядка экспорта может сломать сборку
export * from './types';
export * from './utils';
export * from './enums';
export * from './styles';
export * from './src';
