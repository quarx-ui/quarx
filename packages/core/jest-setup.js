import '@testing-library/jest-dom';
import 'mock-match-media/jest-setup.cjs';
import { setMedia } from 'mock-match-media';
import { cleanup } from '@testing-library/react';

setMedia({
    'prefers-color-scheme': 'light',
});
global.beforeEach(() => {
    cleanup();
    jest.clearAllTimers();
    jest.clearAllMocks();
});
