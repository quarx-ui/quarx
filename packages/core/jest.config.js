module.exports = {
    setupFilesAfterEnv: ['<rootDir>/jest-setup.js'],
    transform: {
        '^.+\\.(js|jsx|ts|tsx)$': [
            'babel-jest',
            {
                rootMode: 'upward',
            },
        ],
    },
    // eslint-disable-next-line no-useless-escape
    modulePathIgnorePatterns: ['.*\.test\.pw\..*'],
    testEnvironment: 'jest-environment-jsdom',
};
