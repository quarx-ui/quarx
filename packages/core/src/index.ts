import './polyfills';

// WARNING: Изменение порядка экспорта может сломать сборку
export * from './types';
export * from './enums';
export * from './utils';
export * from './styles';
export * from './components';
