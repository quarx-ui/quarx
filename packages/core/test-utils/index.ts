export { expectRootCn } from './expectRootCn';
export * from './testBEM';
