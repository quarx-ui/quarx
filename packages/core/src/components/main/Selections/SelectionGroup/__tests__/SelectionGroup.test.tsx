import { render, screen } from '@testing-library/react';
import {
    PALETTE_COLORS,
    QX_SIZE,
    SELECTION_GROUP_TYPE,
    SelectionGroup,
    SelectionGroupProps,
    SelectionGroupStyleParams,
    SelectionList,
} from '@core';
import { testStyleParams } from '@quarx-ui/core/test-utils';

describe('SelectionGroup', () => {
    testStyleParams<Partial<SelectionGroupStyleParams>, SelectionGroupProps>(
        SelectionGroup,
        {
            type: SELECTION_GROUP_TYPE.text,
            size: QX_SIZE.medium,
            color: PALETTE_COLORS.brand,
        },
        {
            children: (
                <SelectionList
                    nodes={[]}
                    onUpdate={() => undefined}
                />
            ),
        },
    )({
        color: Object.values(PALETTE_COLORS),
        size: ['small', 'medium', 'large'],
        type: Object.values(SELECTION_GROUP_TYPE),
    });

    test('text/children should be in the document', () => {
        render((
            <SelectionGroup
                title="Title"
                description="Description"
                helperText="HelperText"
            >
                <span>SelectionGroup</span>
            </SelectionGroup>
        ));
        expect(screen.queryByText('Title')).toBeInTheDocument();
        expect(screen.queryByText('Description')).toBeInTheDocument();
        expect(screen.queryByText('HelperText')).toBeInTheDocument();
        expect(screen.queryByText('SelectionGroup')).toBeInTheDocument();
    });

    test('text should be hidden', () => {
        render((
            <SelectionGroup>
                <span>SelectionGroup</span>
            </SelectionGroup>
        ));
        expect(screen.queryByText('Title')).not.toBeInTheDocument();
        expect(screen.queryByText('Description')).not.toBeInTheDocument();
        expect(screen.queryByText('HelperText')).not.toBeInTheDocument();
    });
});
