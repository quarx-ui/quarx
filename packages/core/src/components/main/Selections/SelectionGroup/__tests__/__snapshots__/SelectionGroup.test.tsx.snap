// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectionGroup style params color brand 1`] = `
<DocumentFragment>
  <div
    class="QxSelectionGroup QxSelectionGroup_color_brand QxSelectionGroup_size_medium QxSelectionGroup_type_text css-833b18-QxSelectionGroup-root"
    data-testid="testStyleParams-color-brand"
  >
    <div
      class="QxSelectionGroup-content css-1orua0s-QxSelectionGroup-content"
    >
      <div
        class="QxSelectionList QxSelectionList_type_text QxSelectionList_size_medium QxSelectionList_color_brand css-e2m2qr-QxSelectionList-root"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`SelectionGroup style params color danger 1`] = `
<DocumentFragment>
  <div
    class="QxSelectionGroup QxSelectionGroup_color_danger QxSelectionGroup_size_medium QxSelectionGroup_type_text css-833b18-QxSelectionGroup-root"
    data-testid="testStyleParams-color-danger"
  >
    <div
      class="QxSelectionGroup-content css-1orua0s-QxSelectionGroup-content"
    >
      <div
        class="QxSelectionList QxSelectionList_type_text QxSelectionList_size_medium QxSelectionList_color_danger css-e2m2qr-QxSelectionList-root"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`SelectionGroup style params color info 1`] = `
<DocumentFragment>
  <div
    class="QxSelectionGroup QxSelectionGroup_color_info QxSelectionGroup_size_medium QxSelectionGroup_type_text css-833b18-QxSelectionGroup-root"
    data-testid="testStyleParams-color-info"
  >
    <div
      class="QxSelectionGroup-content css-1orua0s-QxSelectionGroup-content"
    >
      <div
        class="QxSelectionList QxSelectionList_type_text QxSelectionList_size_medium QxSelectionList_color_info css-e2m2qr-QxSelectionList-root"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`SelectionGroup style params color secondary 1`] = `
<DocumentFragment>
  <div
    class="QxSelectionGroup QxSelectionGroup_color_secondary QxSelectionGroup_size_medium QxSelectionGroup_type_text css-833b18-QxSelectionGroup-root"
    data-testid="testStyleParams-color-secondary"
  >
    <div
      class="QxSelectionGroup-content css-1orua0s-QxSelectionGroup-content"
    >
      <div
        class="QxSelectionList QxSelectionList_type_text QxSelectionList_size_medium QxSelectionList_color_secondary css-e2m2qr-QxSelectionList-root"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`SelectionGroup style params color success 1`] = `
<DocumentFragment>
  <div
    class="QxSelectionGroup QxSelectionGroup_color_success QxSelectionGroup_size_medium QxSelectionGroup_type_text css-833b18-QxSelectionGroup-root"
    data-testid="testStyleParams-color-success"
  >
    <div
      class="QxSelectionGroup-content css-1orua0s-QxSelectionGroup-content"
    >
      <div
        class="QxSelectionList QxSelectionList_type_text QxSelectionList_size_medium QxSelectionList_color_success css-e2m2qr-QxSelectionList-root"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`SelectionGroup style params color warning 1`] = `
<DocumentFragment>
  <div
    class="QxSelectionGroup QxSelectionGroup_color_warning QxSelectionGroup_size_medium QxSelectionGroup_type_text css-833b18-QxSelectionGroup-root"
    data-testid="testStyleParams-color-warning"
  >
    <div
      class="QxSelectionGroup-content css-1orua0s-QxSelectionGroup-content"
    >
      <div
        class="QxSelectionList QxSelectionList_type_text QxSelectionList_size_medium QxSelectionList_color_warning css-e2m2qr-QxSelectionList-root"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`SelectionGroup style params size large 1`] = `
<DocumentFragment>
  <div
    class="QxSelectionGroup QxSelectionGroup_color_brand QxSelectionGroup_size_large QxSelectionGroup_type_text css-833b18-QxSelectionGroup-root"
    data-testid="testStyleParams-size-large"
  >
    <div
      class="QxSelectionGroup-content css-psg98-QxSelectionGroup-content"
    >
      <div
        class="QxSelectionList QxSelectionList_type_text QxSelectionList_size_large QxSelectionList_color_brand css-e2m2qr-QxSelectionList-root"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`SelectionGroup style params size medium 1`] = `
<DocumentFragment>
  <div
    class="QxSelectionGroup QxSelectionGroup_color_brand QxSelectionGroup_size_medium QxSelectionGroup_type_text css-833b18-QxSelectionGroup-root"
    data-testid="testStyleParams-size-medium"
  >
    <div
      class="QxSelectionGroup-content css-1orua0s-QxSelectionGroup-content"
    >
      <div
        class="QxSelectionList QxSelectionList_type_text QxSelectionList_size_medium QxSelectionList_color_brand css-e2m2qr-QxSelectionList-root"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`SelectionGroup style params size small 1`] = `
<DocumentFragment>
  <div
    class="QxSelectionGroup QxSelectionGroup_color_brand QxSelectionGroup_size_small QxSelectionGroup_type_text css-833b18-QxSelectionGroup-root"
    data-testid="testStyleParams-size-small"
  >
    <div
      class="QxSelectionGroup-content css-1i4jph7-QxSelectionGroup-content"
    >
      <div
        class="QxSelectionList QxSelectionList_type_text QxSelectionList_size_small QxSelectionList_color_brand css-e2m2qr-QxSelectionList-root"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`SelectionGroup style params type contained 1`] = `
<DocumentFragment>
  <div
    class="QxSelectionGroup QxSelectionGroup_color_brand QxSelectionGroup_size_medium QxSelectionGroup_type_contained css-17kpwxr-QxSelectionGroup-root"
    data-testid="testStyleParams-type-contained"
  >
    <div
      class="QxSelectionGroup-content css-1orua0s-QxSelectionGroup-content"
    >
      <div
        class="QxSelectionList QxSelectionList_type_contained QxSelectionList_size_medium QxSelectionList_color_brand css-1ksv9kq-QxSelectionList-root"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`SelectionGroup style params type text 1`] = `
<DocumentFragment>
  <div
    class="QxSelectionGroup QxSelectionGroup_color_brand QxSelectionGroup_size_medium QxSelectionGroup_type_text css-833b18-QxSelectionGroup-root"
    data-testid="testStyleParams-type-text"
  >
    <div
      class="QxSelectionGroup-content css-1orua0s-QxSelectionGroup-content"
    >
      <div
        class="QxSelectionList QxSelectionList_type_text QxSelectionList_size_medium QxSelectionList_color_brand css-e2m2qr-QxSelectionList-root"
      />
    </div>
  </div>
</DocumentFragment>
`;
